table.go

```diff
@@ -950,15 +949,25 @@ func newTableBlock(table *Table, prevTx, tx uint64, id ulid.ULID) (*TableBlock,
 		prevTx: prevTx,
 	}
 
-	var err error
+	cfg, recovered, err := table.configureLSMLevels(table.db.columnStore.indexConfig)
+	if err != nil {
+		return nil, fmt.Errorf("configure LSM levels: %w", err)
+	}
+
 	tb.index, err = index.NewLSM(
 		table.name,
 		table.schema,
-		table.configureLSMLevels(table.db.columnStore.indexConfig),
+		cfg,
+		table.db.Wait,
```

Comment:
How does waiting for previous transactions affect throughput/latency? My main concern is that I'm not sure whether we should be blocking write transactions at all.

Reply:
It doesn't because we don't wait for previous transactions to complete a transaction. 

The wait function is used during compaction to ensure we have all transactions lower than the wm in a compacted part.